src/map/skill.cpp

```diff
@@ -7152,6 +7403,18 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 			flag |= 8;
 		if (skill_id == SR_HOWLINGOFLION)
 			starget = splash_target(src);
+		if (skill_id == SJ_NEWMOONKICK)
+			clif_skill_nodamage(src, bl, skill_id, skill_lv, sc_start(src, bl, SC_NEWMOON, 100, skill_lv, skill_get_time(skill_id, skill_lv)));
+		if (skill_id == SJ_STAREMPEROR && sc && sc->data[SC_DIMENSION]) {
+			if (sd) {
+				// Remove old shields if any exist.
+				pc_delspiritball(sd, sd->spiritball, 0);
+				for (i = 0; i < 2; i++)				
+					pc_addspiritball(sd,skill_get_time2(SJ_BOOKOFDIMENSION, 1), 2);
+			}
+			status_change_end(src, SC_DIMENSION, INVALID_TIMER);
```

Comment:
SJ_STAREMPEROR  +  SJ_BOOKOFDIMENSION
will get 2 shield spiritball

shield spiritball can defend damage is MaxSP

Now can view 2 spiritball in client,
But defend damage not work.